### ISSUE TITLE
Mattermost moving to API version 4

### ISSUE THREAD

**Issue by jasonblais** [LPgenerator/mattermost_bot#35]
Those who are following this thread:

@attzonko helped create a new repo from this project with APIv4 support: https://github.com/attzonko/mmpy_bot

Would love to have you contribute and share feedback on how the bot is working!

----

Hey @gotlium !

First, hugely appreciate you sharing this project back with the world. Fantastic work!

I wanted to let you know about our plans moving to API v4: 

To make the Mattermost API web service easier to use and to offer more powerful options for these integrations, Mattermost will be moving to a [new API version](https://docs.google.com/document/d/197JwEBMnK8okFilTfGSpbsrXPY5RZOJ4gG2DXwcbwYE/edit) soon.  Highlights include:  

* Fully documented API endpoints 
* More in-depth access to server functionality 
* Wider use of established HTTP verbs 
* Consistent endpoint structures 
* A new and improved Go driver  

We plan to release API version 4 on March 16th, with Mattermost server 3.7.  While the current API version 3 will be supported until September 16th, we recommend you begin using API version 4 soon after its release.

#### Contributing  

API version 4 is an active and on-going project. If you're interested in helping contribute, [please join our Mattermost community instance](https://pre-release.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676) and the [APIv4 channel](https://pre-release.mattermost.com/core/channels/apiv4). 

We've prepared a [contribution process for APIv4](https://docs.mattermost.com/developer/api4.html) and a [progress tracker for new APIv4 enpoints](https://docs.google.com/spreadsheets/d/1nPoLgwh_9zRFECpqRUZAKIWihCmX27pnDtFGLtG_WnY/edit#gid=0) to help you get started.

We're also open for suggestions on adding new API endpoints to help with your integration.

**Comment by attzonko** [comment]
I am also very interested if there are plans to update this bot for API version 4. I would be willing to help with some guidance :)

**Comment by gotlium** [comment]
Hi, Thanks. I think it's a good idea to migrate to new documented API.

**Comment by diegoroccia** [comment]
Hi
in the meanwhile , does this work with mattermost 3.9 ?

**Comment by attzonko** [comment]
I am running 3.10 and have no issue with this bot. Mattermost is still supporting the older API.

I guess eventually this code would need to be migrated to API V4 but for now it still seems to work.

**Comment by SoundGoof** [comment]
it seems to be a deadline now...

All APIv3 endpoints are scheduled for removal on January 16, 2018.
https://docs.mattermost.com/administration/changelog.html#release-v4-1-0

**Comment by gotlium** [comment]
Hi, unfortunately I quit from this company and no longer supporting this app.

**Comment by jasonblais** [comment]
@gotlium Who is supporting this app now, do you know?

**Comment by gotlium** [comment]
Hi, I don't known. Possible we need to search new contributors.

**Comment by jasonblais** [comment]
Thanks @gotlium for the quick response! 

@attzonko You previously signalled interest helping update this bot to API v4? Would you still be interested?

**Comment by attzonko** [comment]
@jasonblais Absolutely. It is sitting on my TODO list :)

**Comment by jasonblais** [comment]
@attzonko Great! You've also signed up at https://pre-release.mattermost.com, correct? If you have any questions, don't hesitate to ask the community or our staff there as well.

Curious what kind of a timeline would you have in mind for the migration? I'm sure you have lots of other items on your todo list :)

**Comment by gotlium** [comment]
When app will be migrate to API v4 and PR, I can release it to PyPi.

**Comment by jasonblais** [comment]
Perfect. Thanks @gotlium!

**Comment by attzonko** [comment]
@jasonblais Yes I am in the pre-release chat rooms. Now that I put 2 and 2 together and know which Jason this is I can bug you on there. As far as the time frame, it depends. the deadline was Jan 17, but existing instances should continue working as is correct?  If my assumption is correct, my goal is to make incremental progress through Jan. When I initially looked at it, there was not too many APIs being used, so it could be fairly quick.

**Comment by jasonblais** [comment]
Perfect. And that's correct, existing instances will continue to work as usual.

**Comment by attzonko** [comment]
@gotlium @jasonblais I created a PR #55 which adds APIv4 support as well as a few of the other features from @seLain with tweaks/fixes from myself.

**Comment by jasonblais** [comment]
This is awesome. Thank you for the quick turnaround!

**Comment by attzonko** [comment]
@jasonblais The main reason for the quick turnaround is that @seLain had already done most of the work back in September.

**Comment by jasonblais** [comment]
@gotlium Would you have thoughts on the best way to transfer ownership of https://pypi.python.org/pypi/mattermost_bot?

@attzonko is open to assuming ownership of the repo and/or forking it for now.

//cc @deniskabalkin

**Comment by Alexandr92** [comment]
Hello,

So, is there a version now of the mattermost_bot that supports Mattermost API v4 ?

**Comment by jasonblais** [comment]
Those who are following this thread:

@attzonko helped create a new repo from this project with APIv4 support: https://github.com/attzonko/mmpy_bot

Would love to have you contribute and share feedback on how the bot is working!
